src/Libraries/Microsoft.Extensions.Http.Resilience/Hedging/ResilienceHttpClientBuilderExtensions.Hedging.cs

```diff
@@ -88,26 +90,48 @@ public static IStandardHedgingHandlerBuilder AddStandardHedgingHandler(this IHtt
                     Throw.InvalidOperationException("Request message snapshot is not attached to the resilience context.");
                 }
 
-                var requestMessage = snapshot.CreateRequestMessage();
-
-                // The secondary request message should use the action resilience context
-                requestMessage.SetResilienceContext(args.ActionContext);
-
-                // replace the request message
-                args.ActionContext.Properties.Set(ResilienceKeys.RequestMessage, requestMessage);
-
+                // if a routing strategy has been configured but it does not return the next route, then no more routes
+                // are availabe, stop hedging
+                Uri? route;
                 if (args.PrimaryContext.Properties.TryGetValue(ResilienceKeys.RoutingStrategy, out var routingPipeline))
                 {
-                    if (!routingPipeline.TryGetNextRoute(out var route))
+                    if (!routingPipeline.TryGetNextRoute(out route))
                     {
-                        // no routes left, stop hedging
                         return null;
                     }
-
-                    requestMessage.RequestUri = requestMessage.RequestUri!.ReplaceHost(route);
                 }
+                else
+                {
+                    route = null;
+                }
+
+                return async () =>
+                {
+                    Outcome<HttpResponseMessage>? actionResult = null;
+
+                    try
+                    {
+                        var requestMessage = await snapshot.CreateRequestMessageAsync().ConfigureAwait(args.ActionContext.ContinueOnCapturedContext);
+
+                        // The secondary request message should use the action resilience context
+                        requestMessage.SetResilienceContext(args.ActionContext);
+
+                        // replace the request message
+                        args.ActionContext.Properties.Set(ResilienceKeys.RequestMessage, requestMessage);
+
+                        if (route != null)
```

Comment:
The equality operator can be overloaded.
```suggestion
                        if (route is not null)
```
Ditto other places.

Reply:
Done 👍